src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+
+  const story = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        copyState = {};
+        break;
+
+      case 'addProperties':
+        copyState = { ...copyState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete copyState[key];
+        });
+        break;
+
+      default:
+        break;
+    }
+    story.push({ ...copyState });
+  });
```

Comment:
CODE STYLE ISSUE: Your code is very good, but there is a more efficient way to perform the same action in all `switch` cases. Instead of pushing the copy of the state to the array inside the `forEach` loop, you can do it once at the end of each loop iteration. This will make your code cleaner and easier to read.